docs: GitHub Actions CI Integrations setup guide (#932)
platform-platform/monorepo: docs/18_github_actions_ci_integration.md

# CI Integrations GitHub Actions integration.

> Summary of the proposed change

Describe the integration mechanism of the CI Integrations component to the GitHub Actions to automatically push the build data.

# References

> Link to supporting documentation, GitHub tickets, etc.

- [CI integrations](metrics/ci_integrations/docs/01_ci_integration_module_architecture.md)

# Motivation

> What problem is this project solving?

This document describes the configuration process of the CI Integrations to automatically synchronize the GitHub Actions build data to the Metrics Web application.

# Goals

> Identify success metrics and measurable goals.

This document aims the following goals: 

- Explain the CI Integrations configuration process.
- Explain the mechanism of automation build imports using the GitHub Actions and CI Integrations tool.

# Non-Goals

> Identify what's not in scope.

This document does not describe the configuration of building or publishing jobs.

> # Design

Reviewer: ```suggestion
# CI Integration tool configuration
```
I think we are free to adjust template sections as we see fit (it's more descriptive)

> Explain and diagram the technical design
>
> Identify risks and edge cases

> To be able to track the state of the applications under development, we should create `YAML` configuration files for each project that will be used by the CI Integrations tool to sync the build data with the Metrics Web application. Assume we have an `Awesome Project` and an `awesome_actions.yml` workflow file with `Awesome Project Actions` job used to run tests of the `Awesome Project`. If so, let's create a sample configuration file for this project. The configuration should consist of the following parts: 

Reviewer: ```suggestion
To be able to synchronize build data using the CI Integrations tool we should create `YAML` configuration files for each project that will be used by the CI Integrations tool to sync the build data with the Metrics Web application. Assume we have an `Awesome Project` and an `awesome_actions.yml` workflow file with `Awesome Project Actions` job used to run tests of the `Awesome Project`. If so, let's create a sample configuration file for this project. The configuration should consist of the following parts: 
```

maybe we can refer regular CI Integrations tool configuration document and focus on Github Actions nuances? Like storing some sensitive information in secrets and maybe just providing a sample configuration file?

| metrics_project_id       | Public | A firestore document identifier of the project to import data to. |

__*Please, NOTE*__ that the `Secret` values must be stored as [GitHub Secrets](https://docs.github.com/en/free-pro-team@latest/actions/reference/encrypted-secrets). The configuration file can contain this fields as a environment variables. To replace them in the GitHub actions workflow you can use `gettext`  or replace it using the `sed` tool. The example of using the `sed` command provided in the [Metrics Integration Actions](#Metrics-Integration-Actions) section.

Once we've finished creating the configuration file, we should configure the GitHub Actions that will export the build data to the Metrics Web application. 

To export the data, we should configure the following actions: 

- [`Notify about the building project`](#Notify-about-the-building-project) - the job needed to notify the `Metrics Integration Actions` that some project build was started.
- [`Metrics Integration Actions`](#Metrics-Integration-Actions) - the workflow needed to export the data to the Metrics Web application using the CI Integrations component.

Let's consider each action in more detail.

> ## Notify about the building project

Reviewer: What if we put the sequence diagram somewhere around here - so that right after we can explain its steps? Users will be able to see the complete picture first and then explore the steps of interest?

> The `Notify about the building project` step notifies the `Metrics Integration Actions` about some project's build started. This job should emit a repository dispatch event containing `client_payload` with the `building_awesome_project` boolean field in the `client_payload`. To send the repository dispatch event, we are using the [Repository Dispatch](https://github.com/marketplace/actions/repository-dispatch) action.

Reviewer: ```suggestion
The `Notify about the finishing awesome project build` step notifies the `Metrics Integration Actions` about some project's build started. This job should emit a repository dispatch event containing `client_payload` with the `building_awesome_project` boolean field in the `client_payload`. To send the repository dispatch event, we are using the [Repository Dispatch](https://github.com/marketplace/actions/repository-dispatch) action.
```
maybe this name would be more descriptive? Also what if we add a section on why we need this step? Maybe some context like: "To synchronize build data we need to have the current build finished (as CI Integration only synchronizes finished builds) so that means we can't put sync into the same github workflow as a step..."
